### ISSUE TITLE
Option to return nil

### ISSUE THREAD

**Issue by kevintraver** [prodis/wannabe_bool#5]
It would be awesome to have an option to return `nil` if the value is not an acceptable **truthy** or **falsey** value. Or maybe raise an exception?

A use case would be if parsing params, you wouldn't want to always have everything that is not _true_ default to _false_, you might instead want to throw some sort of error if the param was not an acceptable **true/false** input.
```
'true'.to_b        # => true
'false'.to_b       # => false
'invalid'.to_b     # => nil
42.to_b            # => nil
```

Maybe this is something that can be setup in an initializer?

Thanks!

**Comment by kevintraver** [comment]
Or, instead of changing core functionality, maybe add a helper that can check if it's an acceptable *truthy* or *falsey* value.

**Comment by prodis** [comment]
@kevintraver 
Yes, we can implement the setup of these "falsey behaviours" (false, nil or exception) in an initializer. 

But before, we need to specify which values we can consider **falsey**, by type.

My suggestion would be:

- String: `"f"`, `"false"`, `"off"`, `"n"`, `"no"` or `"0"`
- Symbol: the same rule of strings
- Numeric: `0`

Sorry, I didn't understand the second option to add a helper. Can you explain more or exemplify with a pseudocode, please?

Thank you.

**Comment by kevintraver** [comment]
perhaps something like
```
WannabeBool.is_valid(true)                  # => true
WannabeBool.is_valid(false)                 # => false
WannabeBool.is_valid('something invalid')   # => false
```

**Comment by prodis** [comment]
@kevintraver 
Please take a look in this branch: https://github.com/prodis/wannabe_bool/tree/falsey
I would like you validate with your real use case, before I merge.

Possible "falsey behaviours": `:false` (default), `:nil`, and `:error`.

Example:
```ruby
require 'wannabe_bool'

'fsdafas'.to_b # => false

WannabeBool.invalid_value_behaviour = :nil
'fsdafas'.to_b # => nil

WannabeBool.invalid_value_behaviour = :error
'fsdafas'.to_b # => ArgumentError: is not a valid boolean representation
```

And please give your opinion about the name of configuration option `invalid_value_behaviour`. I thought in `falsey_behaviour` too. What do you think? Any another name suggestion?

Thank you.

**Comment by kevintraver** [comment]
Looks good to me. Thanks!

**Comment by prodis** [comment]
@kevintraver 
The final implementation was a little bit different from my last comment. Please take a look in the instructions on README at version 0.60.

Thank you for your issue and suggestion.

**Comment by prodis** [comment]
Solved with version 0.6.0.
